src/main/java/com/kaleido/fetch/service/FetchService.java

```diff
@@ -100,30 +98,28 @@ public FetchService(KaptureClient<Media> mediaKaptureClient, KaptureClient<Batch
     }
 
     public List<Component> getComponentsByClassificationAndId(List<Component> searchComponents) {
-        log.debug("call to /components/find : {}", searchComponents);
-
-        var mediaComponents = CompletableFuture.supplyAsync(() -> searchMediaByIds(
+        var mediaComponents = CompletableFuture.supplyAsync(() -> searchByIds(
```

Comment:
If you're going to use `searchByIds` then it is inefficient to call it 4 times. It would be simpler to call it once and then use the `groupingBy` collector to return a map of `String classification` to `List<Component>`

See for example https://www.baeldung.com/java-groupingby-collector